metagraph_karateclub/plugins/karateclub/algorithms.py

```diff
@@ -0,0 +1,47 @@
+from metagraph import concrete_algorithm
+from .. import has_karateclub
+
+if has_karateclub:
+    import karateclub
+    import numpy as np
+    import networkx as nx
+    from metagraph.plugins.networkx.types import NetworkXGraph
+    from metagraph.plugins.numpy.types import (
+        NumpyMatrix,
+        NumpyNodeMap,
+        NumpyNodeEmbedding,
+    )
+
+    @concrete_algorithm("embedding.train.node2vec")
+    def karateclub_node2vec_train(
+        graph: NetworkXGraph,
+        p: float,
+        q: float,
+        walks_per_node: int,
+        walk_length: int,
+        embedding_size: int,
+        epochs: int,
+        learning_rate: float,
+        worker_count: int = 1,
+    ) -> NumpyNodeEmbedding:
+        trainer = karateclub.Node2Vec(
+            walk_number=walks_per_node,
+            walk_length=walk_length,
+            workers=worker_count,
+            p=p,
+            q=q,
+            dimensions=embedding_size,
+        )
+        old2canonical = {
```

Comment:
Does karate club require nodes to be numbered 0..n? The incoming graph should already have valid NodeIDs for the node labels, although they may not be perfectly sequential.

If this is a requirement, you should make a copy of the graph. Modifying the node ids changes the meaning of the graph, so modifying the input graph directly is wrong.

Reply:
Yes, `karateclub` requires nodes to be numbered sequentially. The code that does that is [here](https://github.com/benedekrozemberczki/karateclub/blob/master/karateclub/estimator.py#L53).

I think `nx.relabel_nodes` makes a copy by default since it has the default parameter `copy=True`. https://github.com/metagraph-dev/metagraph-karateclub/pull/1/commits/149a9a65ddc9bbb84bc910714b7a87f85edce25f makes the copy explicit by explicitly passing in `copy=True`.